airbyte-integrations/connectors/source-zendesk-support/unit_tests/integrations/test_users.py

```diff
@@ -0,0 +1,109 @@
+            .with_subdomain("d3v-airbyte")
+            .with_start_date(pendulum.now(tz="UTC").subtract(years=2))
+        )
+
+    def _get_authenticator(self, config):
+        return ApiTokenAuthenticator(email=config["credentials"]["email"], password=config["credentials"]["api_token"])
+
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_records(self, http_mocker):
+        config = self._config().with_start_date(_START_DATE).build()
+        api_token_authenticator = self._get_authenticator(config)
+        http_mocker.get(
+            UsersRequestBuilder.endpoint(api_token_authenticator).with_include("identities").with_start_time(_START_DATE).build(),
+            UsersResponseBuilder.identities_response()
+            .with_record(UsersRecordBuilder.record())
+            .with_record(UsersRecordBuilder.record())
+            .build(),
+        )
+
+        output = read_stream("user_identities", SyncMode.full_refresh, config)
+
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_next_page_when_read_then_paginate(self, http_mocker):
+        config = self._config().with_start_date(_START_DATE).build()
+        api_token_authenticator = self._get_authenticator(config)
+        http_mocker.get(
+            UsersRequestBuilder.endpoint(api_token_authenticator).with_include("identities").with_start_time(_START_DATE).build(),
+            UsersResponseBuilder.identities_response(UsersRequestBuilder.endpoint(api_token_authenticator).build(), _A_CURSOR)
+            .with_record(UsersRecordBuilder.record())
+            .with_record(UsersRecordBuilder.record())
+            .with_pagination()
+            .build(),
+        )
+        http_mocker.get(
+            UsersRequestBuilder.endpoint(api_token_authenticator).with_include("identities").with_cursor(_A_CURSOR).build(),
+            UsersResponseBuilder.identities_response().with_record(UsersRecordBuilder.record()).build(),
+        )
+
+        output = read_stream("user_identities", SyncMode.full_refresh, config)
+
+        assert len(output.records) == 3
+
+    @HttpMocker()
+    def test_when_read_then_set_state_value_to_most_recent_cursor_value(self, http_mocker):
+        config = self._config().with_start_date(_START_DATE).build()
+        api_token_authenticator = self._get_authenticator(config)
+        most_recent_cursor_value = _START_DATE.add(days=2)
+        http_mocker.get(
+            UsersRequestBuilder.endpoint(api_token_authenticator).with_include("identities").with_start_time(_START_DATE).build(),
+            UsersResponseBuilder.identities_response()
+            .with_record(UsersRecordBuilder.record().with_cursor(datetime_to_string(most_recent_cursor_value)))
+            .build(),
+        )
+
+        output = read_stream("user_identities", SyncMode.full_refresh, config)
+
+        assert output.most_recent_state.stream_state.__dict__ == {"updated_at": str(most_recent_cursor_value.int_timestamp)}
```

Comment:
This is the only change between the previous implementation and the new one i.e. that state goes from ISO8601 to epoch as a string